iOS Client/PodcastsController/PodcastsCoordinator.swift

```diff
@@ -81,4 +104,27 @@ class PodcastsCoordinator: Coordinator, StoreSubscriber {
             player.play()
         }
     }
+    
+    func getCurrentDuration() -> Double {
+        let time = player.currentTime()
+        return time.seconds
+    }
+    
+    // MARK: - AVAudioPlayerDelegate
+    func audioPlayerDidFinishPlaying(_ player: AVAudioPlayer, successfully flag: Bool) {
+        if flag {
+            MBStore.sharedStore.dispatch(FinishedPodcast())
+        } else {
+            MBStore.sharedStore.dispatch(PodcastError())
+        }
+    }
+    
+    func seek(to second: Double) {
```

Comment:
this should be explicitly marked as the `PodcastHandler` method, which should really be called something like `PodcastViewControllerDelegate` if you decide to stick with delegation for the callback. I would say it's a no brainer, but with UDF in the equation I could see a desire to fire an action and let that come back down. Up to you. Delegation seems to be much more direct